ferretdb/ferretdb.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ferretdb
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/FerretDB/FerretDB/internal/clientconn"
+	"github.com/FerretDB/FerretDB/internal/handlers/registry"
+	"github.com/FerretDB/FerretDB/internal/util/logging"
+)
+
+// Config contains a backend connection strings.
+type Config struct {
+	Handler     string
+	PostgresURL string
+	TigrisURL   string
+}
+
+// FerretDB proxy.
+type FerretDB struct {
+	config Config
+}
+
+// New registers backend handler and returns a new FerretDB.
+func New(conf Config) FerretDB {
+	switch conf.Handler {
+	case "pg":
+		registry.RegisterPg(conf.PostgresURL)
+	case "tigris":
+		registry.RegisterTigris(conf.TigrisURL)
+	default:
+		panic(fmt.Sprintf("Unknown backend handler %q.", conf.Handler))
+	}
+	return FerretDB{
+		config: conf,
+	}
+}
+
+// GetConnectionString returns the backend connection string.
+func (fdb *FerretDB) GetConnectionString() string {
+	return "mongodb://127.0.0.1:27017"
+}
+
+// Run runs the FerretDB proxy as a library with:
+// * error level logging
+// * monitoring disabled
+// * handler PostgreSQL.
+func (fdb *FerretDB) Run(ctx context.Context) error {
+	listenAddr := "127.0.0.1:27017"
```

Comment:
Are we going to make it configurable (especially, the port)? Is it for another PR?

Reply:
> Are we going to make it configurable (especially, the port)? Is it for another PR?

No, we are not going to make it configurable, please see the issue description.